### PR TITLE
Added postgres implementation for indexer

### PR DESCRIPTION
**Pull request by habdelra**
This PR introduces a postgres implementation of the indexer. Primarily this works thru smoothing out issues between postgres and sqlite queries. my take on this is to write the SQL as close to postgres as possible, and when there are SQLite incompatibilities to favor addressing the incompatibilities in the SQLiteAdpter as opposed to the PgAdapter.

TODO:
- [x] The tests for the indexer in postres are almost exactly identical to the sqlite tests (the only difference is the DB adapter we use). Let's try to move the tests into a common place so they aren't duplicated

### CODE REVIEW

File: packages/realm-server/migrations/1712771547705_initial.js
Comment (habdelra) [inline packages/realm-server/migrations/1712771547705_initial.js:1]: There is no data to migrate yet so i didn't bother creating a new migration

File: packages/runtime-common/indexer/client.ts [hunk 1]
```diff
@@ -173,15 +170,15 @@ export class IndexerDBClient {
...
+       CROSS JOIN LATERAL jsonb_array_each(i.deps) as deps_each
        INNER JOIN realm_versions r ON i.realm_url = r.realm_url
        WHERE 
-         deps_each.value =`,
+         deps_each.text_value =`,
       param(cardId),
       'AND',
       ...realmVersionExpression({ useWorkInProgressIndex: true }),
-      'ORDER BY i.card_url',
+      'ORDER BY i.card_url COLLATE "POSIX"',
```
Comment (habdelra) [inline packages/runtime-common/indexer/client.ts:181]: this is an interesting quirk in postgres where you need to tell it to sort a URL like a string (it's special URL style sorting was causing differences in test results between sqlite and postgres). honestly we don't care too strongly about URL sorting--just that it's deterministic, and it needs to be consistent between SQLite and postres

File: packages/realm-server/migrations/1712771547705_initial.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ exports.up = (pgm) => {
     types: 'jsonb',
     embedded_html: 'varchar',
     isolated_html: 'varchar',
-    indexed_at: 'integer',
+    indexed_at: 'bigint',
```
Comment (habdelra) [inline packages/realm-server/migrations/1712771547705_initial.js:13]: There is no data to migrate yet so i didn't bother creating a new migration

File: packages/runtime-common/tests/indexer.ts
Comment (habdelra) [inline packages/runtime-common/tests/indexer.ts:1]: these tests are unchanged from when we ran them in SQLite (with the exception of some really minor postgres syntactical sugar)

**Review by jurgenwerk — APPROVED** [review]

**Review by FadhlanR — APPROVED** [review]

**Pull request merged**